test/jasmine/tests/select_test.js

```diff
@@ -0,0 +1,183 @@
+var Plotly = require('@lib/index');
+var Lib = require('@src/lib');
+var DBLCLICKDELAY = require('@src/plots/cartesian/constants').DBLCLICKDELAY;
+
+var createGraphDiv = require('../assets/create_graph_div');
+var destroyGraphDiv = require('../assets/destroy_graph_div');
+var mouseEvent = require('../assets/mouse_event');
+
+
+describe('select box and lasso', function() {
+    var mock = require('@mocks/14.json');
+
+    afterEach(destroyGraphDiv);
+
+    function drag(path) {
+        var len = path.length;
+
+        mouseEvent('mousemove', path[0][0], path[0][1]);
+        mouseEvent('mousedown', path[0][0], path[0][1]);
+
+        path.slice(1, len).forEach(function(pt) {
+            mouseEvent('mousemove', pt[0], pt[1]);
+        });
+
+        mouseEvent('mouseup', path[len - 1][0], path[len - 1][1]);
+    }
+
+    function click(x, y) {
+        mouseEvent('mousemove', x, y);
+        mouseEvent('mousedown', x, y);
+        mouseEvent('mouseup', x, y);
+    }
+
+    function doubleClick(x, y, cb) {
+        click(x, y);
+        setTimeout(function() {
+            click(x, y);
+            cb();
+        }, DBLCLICKDELAY / 2);
+    }
+
+    function assertRange(actual, expected) {
+        var PRECISION = 4;
+
+        expect(actual.x[0]).toBeCloseTo(expected.x[0], PRECISION);
+        expect(actual.x[1]).toBeCloseTo(expected.x[1], PRECISION);
+        expect(actual.y[0]).toBeCloseTo(expected.y[0], PRECISION);
+        expect(actual.y[1]).toBeCloseTo(expected.y[1], PRECISION);
+    }
+
+    describe('select events', function() {
+        var mockCopy = Lib.extendDeep({}, mock);
+        mockCopy.layout.dragmode = 'select';
+
+        var gd;
+        beforeEach(function(done) {
+            gd = createGraphDiv();
+
+            Plotly.plot(gd, mockCopy.data, mockCopy.layout)
+                .then(done);
+        });
+
+        it('should trigger events', function(done) {
```

Comment:
Maybe add some comments as to the specifics of the test. It's not immediately clear what's being tested.